extras/dhcpserver/dhcpserver.c

```diff
@@ -150,6 +173,8 @@ static void dhcpserver_task(void *pxParameter)
             printf("DHCP Server Error: Failed to receive DHCP packet. err=%d\r\n", err);
             continue;
         }
+		
+		taskENTER_CRITICAL();
```

Comment:
Looks like too much is being done in the critical regions. Probably need to use a mutex instead.